python/conf_validation.py

```diff
@@ -0,0 +1,89 @@
+# Entry point to validating NGen catchment,nexus, and realization files
+import sys, json
+from ngen.config.realization import NgenRealization
+from ngen.config.hydrofabric import NGenCatchmentFile, NGenNexusFile
+
+def main(catchment_file,catchment_subset,nexus_file,nexus_subset,rel_file):
+    """
+    validate the three config files and sub selections
+    """
+    # Validate Catchment config
+    with open(catchment_file) as fp:
```

Comment:
For the most part, the rest of my comments can be ignored in this file. I wanted to offer some pointers (no pun intended) that might make your life a little easier and de-duplicate work.

You could re-write the catchment validation like:
```python
serialized_catchments = NGenCatchmentFile.parse_file(catchment_file)

catchments = []
catchment_pairs = []

for feature in serialized_catchments.features:
  id = feature.id
  toid = feature.properties.toid

  catchment_pairs.append((id, toid))
  catchments.append(id)

```

Reply:
You could do the same thing for `NGenNexusFile` too!

Reply:
Took this suggestion too. Makes the code look cleaner and why not use a bell/whistle if we got it!